app/src/main/java/org/jellyfin/mobile/fragment/WebViewFragment.kt

```diff
@@ -157,18 +156,14 @@ class WebViewFragment : Fragment(), NativePlayerHost {
                 val url = request.url
                 val path = url.path?.lowercase(Locale.ROOT) ?: return null
                 return when {
-                    path.endsWith(Constants.WEB_CONFIG_PATH) -> {
-                        runOnUiThread {
-                            webView.evaluateJavascript(JS_INJECTION_CODE) {
-                                onConnectedToWebapp()
-                            }
-                        }
-                        null // continue loading normally
+                    !connected && path.endsWith(Constants.ANY_BUNDLE_PATH) -> {
+                        onConnectedToWebapp()
+                        AssetsPathHandler(context).handle("native/injectionScript.js")
                     }
-                    path.contains("native") -> webView.context.loadAsset("native/${url.lastPathSegment}")
+                    path.contains("/native/") -> AssetsPathHandler(context).handle("native/${url.lastPathSegment}")
                     path.endsWith(Constants.CAST_SDK_PATH) -> {
                         // Load the chrome.cast.js library instead
-                        webView.context.loadAsset("native/chrome.cast.js")
+                        AssetsPathHandler(context).handle("native/chrome.cast.js")
```

Comment:
The `AssetPathHandler` should be a cached, shared instance as a property of the fragment.